R/brood_check.R

```diff
@@ -1350,65 +1400,114 @@ compare_species_parents <- function(Brood_data, Individual_data, approved_list)
 compare_species_brood_parents <- function(Brood_data, Individual_data, approved_list) {
 
   # Find species information of mothers
-  Females <- Brood_data %>%
-    dplyr::filter(!is.na(FemaleID) & !is.na(MaleID)) %>%
-    dplyr::select(Row, PopID, BroodID, FemaleID, BroodSpecies = Species) %>%
+  females <- Brood_data %>%
+    dplyr::filter(!is.na(.data$FemaleID) & !is.na(.data$MaleID)) %>%
+    dplyr::select(.data$Row, .data$PopID, .data$BroodID,
+                  .data$FemaleID, "BroodSpecies" = .data$Species) %>%
     dplyr::left_join(Individual_data[,c("IndvID", "Species")],
                      by=c("FemaleID" = "IndvID")) %>%
-    dplyr::rename(FemaleSpecies = Species)
+    dplyr::rename("FemaleSpecies" = .data$Species)
 
   # Find species information of fathers
-  Males <- Brood_data %>%
-    dplyr::filter(!is.na(FemaleID) & !is.na(MaleID)) %>%
-    dplyr::select(Row, PopID, BroodID, MaleID, BroodSpecies = Species) %>%
+  males <- Brood_data %>%
+    dplyr::filter(!is.na(.data$FemaleID) & !is.na(.data$MaleID)) %>%
+    dplyr::select(.data$Row, .data$PopID, .data$BroodID,
+                  .data$MaleID, "BroodSpecies" = .data$Species) %>%
     dplyr::left_join(Individual_data[,c("IndvID", "Species")],
                      by=c("MaleID" = "IndvID")) %>%
-    dplyr::rename(MaleSpecies = Species)
-
-  # No errors
-  err <- FALSE
-  #error_records <- tibble::tibble(Row = NA_character_)
-  error_output <- NULL
+    dplyr::rename("MaleSpecies" = .data$Species)
 
-  # Warnings
   # Select records where parents are not of the same species as the brood
-  Multi_species_broods <- dplyr::left_join(Females, Males,
+  multi_species_broods <- dplyr::left_join(females, males,
                                            by=c("Row", "PopID", "BroodID", "BroodSpecies")) %>%
-    dplyr::filter(FemaleSpecies != BroodSpecies | MaleSpecies != BroodSpecies)
+    dplyr::filter(.data$FemaleSpecies != .data$BroodSpecies | .data$MaleSpecies != .data$BroodSpecies)
+
+  # Warnings
+  # Common cross-fostering and hybrids are considered "warnings"
+  common_multi_species <- multi_species_broods %>%
+    dplyr::filter((.data$FemaleSpecies == "PARMAJ" & .data$BroodSpecies == "CYACAE") |
+                    (.data$MaleSpecies == "PARMAJ" & .data$BroodSpecies == "CYACAE") |
+                    (.data$FemaleSpecies == "CYACAE" & .data$BroodSpecies == "PARMAJ") |
+                    (.data$MaleSpecies == "CYACAE" & .data$BroodSpecies == "PARMAJ") |
+                    (.data$FemaleSpecies == "FICALB" & .data$BroodSpecies == "FICHYP") |
+                    (.data$MaleSpecies == "FICALB" & .data$BroodSpecies == "FICHYP") |
+                    (.data$FemaleSpecies == "FICHYP" & .data$BroodSpecies == "FICALB") |
+                    (.data$MaleSpecies == "FICHYP" & .data$BroodSpecies == "FICALB"))
```

Comment:
Would it be worth create a data frame with 'known hybrids' or some such? Would prevent you writing this each time.